python/src/aiconfig/editor/client/src/reducers/aiconfigReducer.ts

```diff
@@ -365,6 +217,129 @@ export default function aiconfigReducer(
         },
       };
     }
+    case "RUN_PROMPT_START": {
+      const runningState = setRunningPromptId(dirtyState, action.promptId);
```

Comment:
I don't think it will be significant, but just noting this technically results in an extra object creation/destructured copy (vs just doing a single one that sets dirty and running prompt id simultaneously). Can revisit later if needed

Reply:
Is this an extra creation/destruction from the previous implementation because it's now a function instead of inlined?

<img width="342" alt="Screenshot 2024-01-15 at 15 21 41" src="https://github.com/lastmile-ai/aiconfig/assets/151060367/b8ff66fd-6a5a-48b5-a6d9-2ea5ee4ec6d6">

If so, I'd probably still prefer to do it this way because it's re-used in many different action types and keeps logic centralized and easier to understand